File: java/strategy/DuckSimulator/client/MallardDuck.java
package client;

import behavior.fly.FlyWithWings;
import behavior.quack.Quack;

public class MallardDuck extends Duck {

    public MallardDuck(String name) {
        super(name);
        flyBehavior = new FlyWithWings(name);
        quackBehavior = new Quack(name);
    }
}

File: cpp/strategy/DuckSimulator/client/MallardDuck.cpp
#include "Duck.cpp"

class MallardDuck : public Duck {

public:
    explicit MallardDuck(string duckName) : Duck(duckName) {
        flyBehavior = new FlyWithWings(duckName);
        quackBehavior = new Quack(duckName);
    }
};

